src/core/shape_inference/include/space_to_batch_shape_inference.hpp

```diff
@@ -15,75 +15,72 @@ namespace ov {
+    using TVal = typename TShape::value_type::value_type;
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 4);
 
     const auto& data_shape = input_shapes[0];
     const auto& block_shape = input_shapes[1];
     const auto& pads_begin_shape = input_shapes[2];
     const auto& pads_end_shape = input_shapes[3];
-    const ov::Rank data_rank = data_shape.rank();
-    bool got_const_data = false;
 
     auto inputs_same_ps = pads_begin_shape;
-    NODE_VALIDATION_CHECK(op,
-                          T::merge_into(inputs_same_ps, pads_end_shape) && T::merge_into(inputs_same_ps, block_shape),
-                          "block_shape, pads_begin and pads_end inputs must have the same shape. Got: ",
-                          block_shape,
-                          ", ",
-                          pads_begin_shape,
-                          " and ",
-                          pads_end_shape);
+    NODE_VALIDATION_CHECK(
+        op,
+        TShape::merge_into(inputs_same_ps, pads_end_shape) && TShape::merge_into(inputs_same_ps, block_shape),
+        "block_shape, pads_begin and pads_end inputs must have the same shape. Got: ",
+        block_shape,
+        ", ",
+        pads_begin_shape,
+        " and ",
+        pads_end_shape);
 
     NODE_VALIDATION_CHECK(op,
                           inputs_same_ps.rank().compatible(1),
                           "block_shape and pads inputs must have rank 1. Got: ",
                           inputs_same_ps.rank());
 
-    if (data_rank.is_static()) {
+    if (data_shape.rank().is_static()) {
+        static constexpr size_t spatial_dim_offset = 1;
         NODE_VALIDATION_CHECK(op,
-                              (data_shape.size() >= 2),
+                              (data_shape.size() > spatial_dim_offset),
                               "The data tensor with rank lower than 2 is not supported (data rank: ",
                               data_shape.size(),
                               ")");
 
-        std::vector<int64_t> block_val, pads_begin_val, pads_end_val;
-
-        auto& output_shape = output_shapes[0];
-        output_shape.resize(data_shape.size());
-        if (get_data_as_int64<T>(1, op, block_val, constant_data) &&
-            get_data_as_int64<T>(2, op, pads_begin_val, constant_data) &&
-            get_data_as_int64<T>(3, op, pads_end_val, constant_data)) {
-            got_const_data = true;
-            int64_t block_prod =
-                std::accumulate(begin(block_val), end(block_val), int64_t(1), std::multiplies<int64_t>());
+        auto out_shape = data_shape;
+        std::vector<int64_t> block, pads_begin, pads_end;
+        if (get_data_as_int64<TShape>(1, op, block, constant_data) &&
+            get_data_as_int64<TShape>(2, op, pads_begin, constant_data) &&
+            get_data_as_int64<TShape>(3, op, pads_end, constant_data)) {
+            TVal block_prod = std::accumulate(begin(block), end(block), 1, std::multiplies<int64_t>());
```

Comment:
If `signed` is not considered identical to `int64_t` by some compilator it might break compilation. If guess VC doesn't like that.

Reply:
It looks like there is no complain from compilers on CI, so leave it. Will add cast if required.